# Fix for building 64 bit.
# Need to determine if these changes are still applicable to the latest version,
# and if so, then submit them upstream. 
--- sqlite-autoconf-3310000/sqlite3.c
+++ sqlite-autoconf-3310000/sqlite3.c
@@ -33748,7 +33748,11 @@ static struct unix_syscall {
 #define osFchmod    ((int(*)(int,mode_t))aSyscall[14].pCurrent)
 
 #if defined(HAVE_POSIX_FALLOCATE) && HAVE_POSIX_FALLOCATE
+#if !defined(_LP64) && _FILE_OFFSET_BITS == 64
+  { "fallocate",    (sqlite3_syscall_ptr)posix_fallocate64,  0 },
+#else
   { "fallocate",    (sqlite3_syscall_ptr)posix_fallocate,  0 },
+#endif
 #else
   { "fallocate",    (sqlite3_syscall_ptr)0,                0 },
 #endif
